File: Cpp/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        int result = 0;
        int current = num;

        while(current != 0){
            if(current & 1) current -= 1;
            else current >>= 1;
            ++result;
        }

        return result;
    }
};

File: Kotlin/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.kt
class Solution {
    fun numberOfSteps(num: Int): Int {
        var result = 0
        var current = num

        while(current != 0){
            if(current and 1 == 1) --current
            else current = current shr 1
            ++result
        }

        return result
    }
}